### PR TITLE
Move Type.arrayOf to typesem

### PR DESCRIPTION
**Pull request by RazvanN7**

### CODE REVIEW

File: compiler/src/dmd/cxxfrontend.d [hunk 1]
```diff
@@ -475,6 +475,12 @@ bool equivalent(Type src, Type t)
     return dmd.typesem.equivalent(src, t);
 }
 
+Type arrayOf(Type type)
```
Comment (ibuclaw) [inline compiler/src/dmd/cxxfrontend.d:478]: Would be good to add a link test to cxxfrontend.cc.  I think you are also missing a header entry to `mtype.h`.

**Review by ibuclaw — DISMISSED** [review]

File: compiler/src/dmd/cxxfrontend.d (hunk 1 above)
Comment (RazvanN7) [reply compiler/src/dmd/cxxfrontend.d:478]: Done.

**Comment by thewilsonator** [comment]
@RazvanN7 please rebase

**Comment by RazvanN7** [comment]
@thewilsonator Done.

**Review by thewilsonator — APPROVED** [review]

**Comment by ibuclaw** [comment]
This PR caused a regression:
```
ld: d/expr.o: in function `ExprVisitor::visit(VarExp*)':
gcc/d/expr.cc:2175: undefined reference to `Type::arrayOf()'
ld: d/expr.o: in function `ExprVisitor::visit(ArrayLiteralExp*)':
gcc/d/expr.cc:2574: undefined reference to `Type::sarrayOf(unsigned long)'
...
```

**Comment by RazvanN7** [comment]
I think I messed up a rebase and somehow did not delete the Type members.

Here's the fix:
https://github.com/dlang/dmd/pull/16292

**Pull request merged**